operator_learning_modules/zpk/zpk_operator_learning.py

```diff
@@ -148,32 +166,338 @@ def __init__(self, learned_operators, domain_name, llm):
+
+            for a in self._llm_ops:
+                assert len(set([o.name for o in self._llm_ops[a]])) == len(self._llm_ops[a]), 'operator names are not all different'
+        ################################################################################################################
+
+        ### Call LNDR: Check whether we have NDRs that need to be relearned
+        for action_predicate in self._fits_all_data:
+            if not self._fits_all_data[action_predicate]:
+                transition_for_action = self._transitions[action_predicate]
+                
+                # This is used to prioritize samples in the learning batch
+                def get_batch_probs(data):
+                    assert False, "Assumed off"
+
+                # Initialize from previous set?
+                if action_predicate in self._ndrs and \
+                    ac.zpk_initialize_from_previous_rule_set[self._domain_name]:
+                    init_rule_sets = {action_predicate : self._ndrs[action_predicate]}
+                else:
+                    init_rule_sets = None
+
+                # max explain_examples_transitions
+                max_ee_transitions = ac.max_zpk_explain_examples_transitions[self._domain_name]
+
+                learned_ndrs = learn_ndrs({action_predicate : transition_for_action},
+                    max_timeout=ac.max_zpk_learning_time,
+                    max_action_batch_size=ac.max_zpk_action_batch_size[self._domain_name],
+                    get_batch_probs=get_batch_probs,
+                    init_rule_sets=init_rule_sets,
+                    rng=self._rand_state,
+                    max_ee_transitions=max_ee_transitions,
+                )
+                ndrs_for_action = learned_ndrs[action_predicate]
+                self._ndrs[action_predicate] = ndrs_for_action
+
+                self._fits_all_data[action_predicate] = True
+                is_updated = True 
+        ################################################################################################################
+
+        ### Update all learned and planning operators
+        if is_updated:
+            self._planning_operators.clear()
+            self._learned_operators.clear()
+
+            for action_pred in self._llm_ops:
+                self._planning_operators.update(self._llm_ops[action_pred])
+
+            for act_pred in self._ndrs:
+                for i, ndr in enumerate(self._ndrs[act_pred]):
+                    suffix = len(self._llm_ops[act_pred]) + i
+                    operator = ndr.determinize(name_suffix=suffix)
+                    # No point in adding an empty effect or noisy effect operator
+                    if len(operator.effects.literals) == 0 or NOISE_OUTCOME in operator.effects.literals:
+                        continue
+                    self._learned_operators.add(operator)
+                    self._planning_operators.add(operator)
```

Comment:
i see now - so the idea is that the planning operators is always just the learned operators + LLM ops right? That makes sense and is what I would've expected

Reply:
Yes, that's what it is.